### ISSUE TITLE
infinite recursion in TestAwtPanels

### ISSUE THREAD

**Issue by stephengold** [jMonkeyEngine/jmonkeyengine#1354]
Testing 3.3.1 release candidate (Hash: a150deb) on Linux Mint, running the `TestAwtPanels` app:
```text
Exception in thread "main" Exception in thread "main" java.lang.StackOverflowError
	at sun.reflect.GeneratedConstructorAccessor1.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.lang.Class.newInstance(Class.java:442)
	at com.jme3.system.JmeDesktopSystem.newContextCustom(JmeDesktopSystem.java:257)
	at com.jme3.system.JmeDesktopSystem.newContext(JmeDesktopSystem.java:285)
	at com.jme3.system.JmeSystem.newContext(JmeSystem.java:159)
	at com.jme3.system.awt.AwtPanelsContext.create(AwtPanelsContext.java:223)
	at com.jme3.system.awt.AwtPanelsContext.create(AwtPanelsContext.java:225)
	at com.jme3.system.awt.AwtPanelsContext.create(AwtPanelsContext.java:225)
	at com.jme3.system.awt.AwtPanelsContext.create(AwtPanelsContext.java:225)
```
...and so on for many screenfuls

**Comment by stephengold** [comment]
I verified that this test worked fine in JME 3.2.4 .

**Comment by stephengold** [comment]
It also worked fine in JME 3.3.0-stable, so this is a recent regression. Suspicion immediately falls on 88c9371 .

**Comment by stephengold** [comment]
Now I notice that 88c9371 was never cherry-picked into the v3.3 branch, so that's not it. Now the prime suspect is PR #1336 . TestAwtPanels worked at e1d5e3c ...

**Comment by stephengold** [comment]
... and `TestAwtPanels` fails at 84912d5 , so PR 1336 is definitely the cause. I'm vague about how AWT is supposed to work. @MeFisto94 would you please lend a hand here?

**Comment by MeFisto94** [comment]
I can't imagine how this could ever happen... But I will look into it.

**Comment by MeFisto94** [comment]
Things getting weird.
A) We have an AWTPanelsContext and an AWTContext, the first one seems to be done without knowing the latter?

https://github.com/jMonkeyEngine/jmonkeyengine/blob/e3161f4622abdec3ae06dc72ce11a6cf9488b598/jme3-desktop/src/main/java/com/jme3/system/awt/AwtPanelsContext.java#L239-L245

AwtPanelsContext seems to wrap a "real" context.
The catch here? 84912d5 prevents `LWJGL2` from being passed to https://github.com/jMonkeyEngine/jmonkeyengine/blob/e3161f4622abdec3ae06dc72ce11a6cf9488b598/jme3-desktop/src/main/java/com/jme3/system/JmeDesktopSystem.java#L242-L262

In our case, the renderer seems to be CUSTOMAwtPanelsContext and thus the line was there to set the internal renderer to lwjgl2.

This still does not solve #1336 
@pspeed42  how would you think one should pass the desired LWJGL Version in that case?

My stupidity is also visible here: https://github.com/jMonkeyEngine/sdk/blob/51f9344eb86ddbc3a07824e9af20e3cfc749e3c3/jme3-core/src/com/jme3/gde/core/editor/SceneApplication.java#L27-L31

So, I guess since this would require API Changes of a bigger dimension, this won't make it into 3.3.1

**Comment by stephengold** [comment]
Fixed in v3.3 branch at a667f5e . Currently unresolved in master branch.

**Comment by stephengold** [comment]
Fixed in v3.3 branch at a667f5e . Currently unresolved in master branch.

**Comment by stephengold** [comment]
Fixed in v3.3 branch at a667f5e . Currently unresolved in master branch.

**Comment by stephengold** [comment]
Fixed in v3.3 branch at a667f5e . Currently unresolved in master branch.

**Comment by stephengold** [comment]
Fixed in master branch at c0692df207ee1d61215618c652b3f0c99450f968
